### PR TITLE
List.traverseValidationA and List.sequenceValidationA now preserve the order of errors

### PR DESCRIPTION
**Pull request by ursenzler**
## Proposed Changes

Preserve the order of errors when executing List.traverseValidationA - and as a consequence for List.sequenceValidationA.
Fixes [Should List.sequenceValidationA preserve order? #191](https://github.com/demystifyfp/FsToolkit.ErrorHandling/issues/191)

## Types of changes
- [x] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

Could however be a breaking change because it changes the behaviour!

## Checklist

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your code._

- [x] Build and tests pass locally
- [ ] I have added tests that prove my fix is effective or that my feature works (if appropriate)
- [ ] I have added necessary documentation (if appropriate)

## Further comments

I think this is rather straight forward :-) 

Let me know if something needs to be changed. E.g. I didn't format it with Fantomas because other parts of the file would have been changed as well.

### CODE REVIEW

File: src/FsToolkit.ErrorHandling/List.fs [hunk 1]
```diff
@@ -106,7 +106,12 @@ module List =
             | Ok _, Error errs
             | Error errs, Ok _ -> traverseValidationA' (Error errs) f xs
 
-    let traverseValidationA f xs = traverseValidationA' (Ok []) f xs
+    let traverseValidationA f xs =
```
Comment (TheAngryByrd) [inline src/FsToolkit.ErrorHandling/List.fs:109]: I think there's a similar fix for this type of behavior where you can keep it in `traverseValidationA'`

https://github.com/demystifyfp/FsToolkit.ErrorHandling/blob/31784fa747d7a7dfd5a03c61f8a3ef436c117d51/src/FsToolkit.ErrorHandling/List.fs#L57

**Review by TheAngryByrd — CHANGES_REQUESTED** [review]
Minor Change but looks good otherwise.

**Comment by TheAngryByrd** [comment]
> I didn't format it with Fantomas because other parts of the file would have been changed as well.

That's fine. I'm used to it by now

File: src/FsToolkit.ErrorHandling/List.fs (hunk 1 above)
Comment (ursenzler) [reply src/FsToolkit.ErrorHandling/List.fs:109]: Yes, that is simpler.

**Comment by ursenzler** [comment]
I formatted List.fs with Fantomas, but it's not in an accepted format. I'm a bit lost here.

**Comment by TheAngryByrd** [comment]
hmm did you run `build.cmd/sh formatcode`  or did you run it from your editor? I don't know if VS or Rider pickup the dotnet tool correctly, which the FAKE build is using the dotnet-tool.

**Comment by TheAngryByrd** [comment]
Thank you!

**Comment by ursenzler** [comment]
Thanks for this excellent library!!

**Pull request merged**